src/step/update_system_variables_springs.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "assemble_residual_vector_springs.hpp"
+#include "assemble_stiffness_matrix_springs.hpp"
+#include "step_parameters.hpp"
+
+#include "src/elements/springs/springs.hpp"
+#include "src/state/state.hpp"
+#include "src/system/springs/calculate_distance_components.hpp"
+#include "src/system/springs/calculate_force_coefficients.hpp"
+#include "src/system/springs/calculate_force_vectors.hpp"
+#include "src/system/springs/calculate_length.hpp"
+#include "src/system/springs/calculate_stiffness_matrix.hpp"
+#include "src/system/springs/update_node_state.hpp"
+
+namespace openturbine {
+
+inline void UpdateSystemVariablesSprings(const Springs& springs, State& state) {
+    // Update the displacements from state -> element nodes
+    Kokkos::parallel_for(
+        springs.num_elems,
+        springs::UpdateNodeState{springs.node_state_indices, springs.u1, springs.u2, state.q}
+    );
+
+    // Calculate system variables and perform assembly
+    Kokkos::parallel_for(
+        springs.num_elems,
+        KOKKOS_LAMBDA(const size_t i_elem) {
+            // Calculate the relative distance vector between the two nodes
+            springs::CalculateDistanceComponents{
+                springs.x0, springs.u1, springs.u2, springs.r
+            }(static_cast<int>(i_elem));
```

Comment:
Make these function objects consistent with the ones from beams and masses - they should take `i_elem` as the first argument of the constructor and the `operator()` shouldn't take an argument.